### PR TITLE
ODBC SSL Compliance Fix

### PR DESCRIPTION
**Pull request by forestmvey**
### Description
Fixed ODBC Driver Issues:

- Driver ignores the UseSSL flag
- SQL plugin query fails if opensearch instance has SSL enabled, and host url doesn't use `https://` prefix
- UI allows for mismatch between UseSSL flag, and specified protocol in host url
- Driver defaults to OpenDistro sql endpoint in event of error with SQL plugin query
 
### Issues Resolved
Issue: [280](https://github.com/opensearch-project/sql/issues/280)
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: sql-odbc/src/sqlodbc/dlg_specific.h [hunk 1]
```diff
@@ -120,7 +120,7 @@ extern "C" {
 #define DEFAULT_BOOLSASCHAR 0
 #define DEFAULT_UNIQUEINDEX 1 /* dont recognize */
 #define DEFAULT_LOGLEVEL OPENSEARCH_WARNING
-#define DEFAULT_USE_SSL 0
+#define DEFAULT_USE_SSL 1
```
Comment (penghuo) [inline sql-odbc/src/sqlodbc/dlg_specific.h:123]: Is there any doc should be change accordingly?

Comment (forestmvey) [reply sql-odbc/src/sqlodbc/dlg_specific.h:123]: I could not find any reference to a default value for the use of SSL. We will make a ticket to update this page with at least a reference to download the driver.
https://opensearch.org/docs/latest/search-plugins/sql/odbc/

Comment (acarbonetto) [reply sql-odbc/src/sqlodbc/dlg_specific.h:123]: And also to note the defaults for the ODBC driver.

**Review by penghuo — APPROVED** [review]
Thanks for the change!

**Comment by forestmvey** [comment]
Demo video for this fix: [Issue-280](https://github.com/opensearch-project/sql/issues/280#issuecomment-1165693199)

**Review by acarbonetto — APPROVED** [review]

**Pull request merged**